homeassistant/components/automation/numeric_state.py

```diff
@@ -66,3 +46,48 @@ def state_automation_listener(entity, from_s, to_s):
         hass, entity_id, state_automation_listener)
 
     return True
+
+
+def if_action(hass, config, action):
+    """ Wraps action method with state based condition. """
+
+    entity_id = config.get(CONF_ENTITY_ID)
+
+    if entity_id is None:
+        _LOGGER.error("Missing configuration key %s", CONF_ENTITY_ID)
+        return action
+
+    below = config.get(CONF_BELOW)
+    above = config.get(CONF_ABOVE)
+
+    if below is None and above is None:
+        _LOGGER.error("Missing configuration key."
+                      " One of %s or %s is required",
+                      CONF_BELOW, CONF_ABOVE)
+        return action
+
+    def state_if():
+        """ Execute action if state matches. """
+
+        state = hass.states.get(entity_id)
+        if state is None or _in_range(state.state, above, below):
```

Comment:
Let's remove `state is None` here. It will cause the if check to succeed if the state is not defined while it should not because a not defined state is not in range.